meilisearch/src/routes/indexes/documents.rs

```diff
@@ -574,6 +575,82 @@ pub async fn delete_documents_by_filter(
     Ok(HttpResponse::Accepted().json(task))
 }
 
+#[derive(Debug, Deserr)]
+#[deserr(error = DeserrJsonError, rename_all = camelCase, deny_unknown_fields)]
+pub struct DocumentEditionByFunction {
+    #[deserr(default, error = DeserrJsonError<InvalidDocumentFilter>)]
+    pub filter: Option<Value>,
+    #[deserr(default, error = DeserrJsonError<InvalidDocumentEditionContext>)]
+    pub context: Option<Value>,
+    #[deserr(error = DeserrJsonError<InvalidDocumentEditionFunctionFilter>, missing_field_error = DeserrJsonError::missing_document_edition_function)]
+    pub function: String,
+}
+
+pub async fn edit_documents_by_function(
+    index_scheduler: GuardedData<ActionPolicy<{ actions::DOCUMENTS_ALL }>, Data<IndexScheduler>>,
```

Comment:
What do you think about these actions @ManyTheFish?
Should we:
- Create a new `DOCUMENT_EDIT` action?
- Make the actions a set?

Reply:
> Create a new DOCUMENT_EDIT action?

I don't think so, moreover, it conflicts with DOCUMENT_ADD when using the PUT /document which can be considered as a document edition.

> Make the actions a set?

It would be better because Document::All matches Document::Delete, Document::Add and Document::Get but (Document::Delete + Document::Add + Document::Get) don't match Document::All.
We can keep it like this for now and enhance it later because it would not be breaking. Moreover, it's an experimental feature.

related to https://github.com/meilisearch/meilisearch/issues/4645